### PR TITLE
Add meson build libstartup-notification support

### PR DESCRIPTION
**Pull request by zhuyaliang**
Fix https://github.com/mate-desktop/mate-desktop/issues/570

### CODE REVIEW

**Review by lukefromdc — CHANGES_REQUESTED** [review]
Startup notification isn't working yet with this. A meson build using
`meson setup build --prefix /usr -Dstartup-notification=true`
built and installed fine, but I did not get the busy cursor on starting up a capplet from mate-control-center. I get the busy cursor on capplet startup from m-c-c reliably with the autotools build.

**Comment by zhuyaliang** [comment]
@lukefromdc 
Run this command, and if the return result can be obtained, it indicates that ```startup-notification``` support has been added
```
strings build/libmate-desktop/libmate-desktop-2.so.17.1.4 | grep sn_launcher_context_new 
```

**Comment by lukefromdc** [comment]
On running that, I get

```
sn_launcher_context_new
sn_launcher_context_new
sn_launcher_context_new

```
showing an attempt to build libstartup notification support, but it doesn't actually function at runtime

**Comment by zhuyaliang** [comment]
Check if the Mason installation path is correct and if the installed dynamic library is loading correctly
```
[test@archlinux mate-desktop]$ md5sum build/libmate-desktop/libmate-desktop-2.so.17.1.4
f7e8e2fef93f2716bf9cc5af7cb9a338  build/libmate-desktop/libmate-desktop-2.so.17.1.4
[test@archlinux mate-desktop]$ ldd /usr/bin/mate-control-center | grep libmate-desktop
	libmate-desktop-2.so.17 => /usr/lib/libmate-desktop-2.so.17 (0x0000007fef9d8000)
[test@archlinux mate-desktop]$ md5sum /usr/lib/libmate-desktop-2.so.17
f7e8e2fef93f2716bf9cc5af7cb9a338  /usr/lib/libmate-desktop-2.so.17
[test@archlinux mate-desktop]$ 
```
I am using ```Archlinux``` and this issue has been fixed in my system
@raveit65 Can you help verify if the problem has been resolved

**Comment by raveit65** [comment]
I do no use meson for my fedora rpm packages. Do i have only  re-compile mate-desktop package with meson?
Or whole the mate stack which really want to avoid?
For fedora i have disable some startup notifications, because you see only a spinning cursor but you don't know what is hanging. https://src.fedoraproject.org/rpms/caja/blob/rawhide/f/caja.spec#_100 
The concept of startup-notifications is a bit questionable.

**Comment by raveit65** [comment]
Btw, i don't see any spinning cursor from startup-notification when i start a capplet from mcc.
All is compiled with autotools.

**Comment by lukefromdc** [comment]
I really don't know a lot about this package. Perhaps it commonly breaks? All it seems to do is provide that busy cursor. I would be OK with removing it entirely from mate-desktop if it is unreliable but do rather like it so now that I know about it I build with it.

**Comment by raveit65** [comment]
offtopic on
I need to change my rpm to meson but i didn't find all equivalent build flags with meson.
     --enable-gtk-doc             -->  -Dgtk-doc=true
     --with-pnp-ids-path="%{_datadir}/hwdata/pnp.ids"   -->    -Dpnp-ids-path="%{_datadir}/hwdata/pnp.ids"
     --enable-introspection=yes   -->   -Dintrospection=true
     --disable-static    -->   --default-library=shared 

-Dstartup-notification=true      --> this is new in my rpm, but compiling startup-notifications was always enable in my rpm.
     
But for this build flags i don't find an equivalent ?
     --enable-gtk-doc-html   
     --disable-schemas-compile  
     --with-x

offtopic off

**Comment by raveit65** [comment]
Now with package rpm build with meson.
With or without this branch i see a busy cursor at session startup but not when starting an application via capplet from m-c-c window. Maybe the application starts to fast to see a busy cursor?
Honestly, i have no idea to verify this PR.

**Comment by raveit65** [comment]
Meson build path is OK.
```
[root@mother rave]# ldd /usr/bin/mate-control-center | grep libmate-desktop
	libmate-desktop-2.so.17 => /lib64/libmate-desktop-2.so.17 (0x00007f511c229000)
[root@mother rave]# md5sum /lib64/libmate-desktop-2.so.17
70a34f0e25f79d07ae7ae29186f0644e  /lib64/libmate-desktop-2.so.17
```
```
[root@mother rave]# strings /lib64/libmate-desktop-2.so.17 | grep sn_launcher_context_new
sn_launcher_context_new
```
Seems Ok or not?

**Comment by raveit65** [comment]
In m-c-c i found a build flag which i don't use usually.
`--enable-appindicator=[no/auto/yes]`
Needs this to be enable to see a busy cursor?

**Comment by raveit65** [comment]
Using `--enable-appindicator=yes` doesn't make any different.
I do not see a busy cursor when starting a capplet from mcc.

**Comment by lukefromdc** [comment]
I simply don't know enough about how meson works to diagnose this.

**Review by raveit65 — APPROVED** [review]
I can't verify the issue in result is can't verify the fix.
But fix lgtm.

**Review by lukefromdc — COMMENTED** [review]
Converting my review to a non-blocking comment.

**Comment by zhuyaliang** [comment]
@yetist Please help review the code

File: meson.build [hunk 1]
```diff
@@ -60,6 +60,10 @@ config_h = configuration_data()
 config_h.set_quoted('GETTEXT_PACKAGE', matedt_gettext_package)
 config_h.set_quoted('LOCALEDIR', matedt_localedir)
 
+if get_option('startup-notification')
```
[the remark this answers is not in the document]
Comment (zhuyaliang) [reply meson.build]: done

**Review by lukefromdc — APPROVED** [review]
This now works as intended, and automatically detects libstartup-notification if installed and uses it. Once installed, I now get the busy cursor. Behavior with this installed now matches that of autotools builds.

THANKS! This clears a roadblock in converting MATE in general autotools->meson in that we no longer lose a feature in doing so

**Comment by lukefromdc** [comment]
Holding off on merging for the pending review

**Review by yetist — APPROVED** [review]
LGTM

**Pull request merged**